doc/specs/Configuration object for profiles.md

```diff
@@ -0,0 +1,125 @@
+the parent of the child during a settings reload without the overrides being erased (since those overrides live in the child).
+
+The idea behind unfocused appearance configurations is similar. We will pass in another `TerminalSettings` object to the control,
+which is simply a child that already has some overrides in it. When the control gains or loses focus, it simply switches between
+the two settings objects appropriately.
+
+### Allowed parameters
+
+For now, these states are meant to be entirely appearance-based. So, not all parameters which can be
+defined in a `Profile` can be defined in this new object (for example, we do not want parameters which
+would cause a resize in this object.) Here is the list of parameters we will allow:
+
+- Anything regarding colors: `colorScheme`, `foreground`, `background`, `cursorColor` etc
+- Anything regarding background image: `path`, `opacity`, `alignment`, `stretchMode`
+- `cursorShape`
+
+We may wish to allow further parameters in these objects in the future (like `bellStyle`?). The addition
+of further parameters can be discussed in the future and is out of scope for this spec.
+
+### Inheritance
+
+The inheritance model can be thought of as an 'all-or-nothing' approach in the sense that the `unfocusedConfig` object
+is considered as a *single* setting instead of an object with many settings. We have chosen this model because it is cleaner
+and easier to understand than the alternative, where each setting within an `unfocusedConfig` object has a parent from which
+it obtains its value.
+
+Note that when `TerminalApp` initializes a control, it creates a `TerminalSettings` object for that profile and passes the
+control a child of that object (so that the control can store overrides in the child, as described earlier). If an unfocused
+config is defined in the profile (or in globals/profile defaults), then `TerminalApp` will create a *child of that child*,
+put the relevant overrides in it, and pass that into the control as well. Thus, the inheritance of any undefined parameters
+in the unfocused config will be as follows:
+
+1. The unfocused config specified in the profile (or in globals/profile defaults)
+2. Overrides made by term control
+3. The parent profile
+
+## UI/UX Design
+
+Users will be able to add a new setting to their profiles that will look like this:
+
+```
+"unfocusedConfig":
+{
+    "colorScheme": "Campbell",
+    "cursorColor": "#888",
+    "cursorShape": "emptyBox",
+    "foreground": "#C0C0C0",
+    "background": "#000000"
+}
+```
+
+When certain appearance settings are changed via OSC sequences (such as the background color), only the focused/regular
+appearance will change and the unfocused one will remain unchanged. However, since the unfocused settings object inherits
+from the regular one, it will still apply the change (provided it does not define its own value for that setting).
+
+## Capabilities
+
+### Accessibility
+
+Does not affect accessibility.

```

Comment:
Not directly. I guess a user could choose for themselves some unfocused states that would make the UI highly inaccessible to themselves (poor contrast ratio.)

What happens for high-contrast mode? Is the user allowed to still override high-contrast mode colors to the point where they might not be high-contrast anymore?

Reply:
If the user can override high-contrast mode colors already in their current profiles, then yes they can do so in the unfocused appearance as well. I.e. this change doesn't provide _additional_ functionality with regards to a specific appearance, it just gives users a second appearance for their profiles, so any accessibility concerns with regards to users messing around with their unfocused appearance already exist because they could do the same with the one appearance they have currently.